File: python/MouseWireless.py
# MouseWireless.py
from Mouse import Mouse

class MouseWireless(Mouse):
    def __init__(self, kapasitas_baterai=0, daya_tahan_baterai="", tipe_koneksi="", range_koneksi="", nama="", bentuk_mouse="", DPI="", warna="", jumlah_tombol=0, tipe_koneksi_mouse=""):
        super().__init__(nama, bentuk_mouse, DPI, warna, jumlah_tombol, tipe_koneksi_mouse)
        self.kapasitas_baterai = kapasitas_baterai
        self.daya_tahan_baterai = daya_tahan_baterai
        self.tipe_koneksi = tipe_koneksi
        self.range_koneksi = range_koneksi
    
    def setKapasitasBaterai(self, kapasitas_baterai):
        self.kapasitas_baterai = kapasitas_baterai
    
    def setDayaTahanBaterai(self, daya_tahan_baterai):
        self.daya_tahan_baterai = daya_tahan_baterai
    
    def setTipeKoneksi(self, tipe_koneksi):
        self.tipe_koneksi = tipe_koneksi
    
    def setRangeKoneksi(self, range_koneksi):
        self.range_koneksi = range_koneksi
    
    def getKapasitasBaterai(self):
        return self.kapasitas_baterai
    
    def getDayaTahanBaterai(self):
        return self.daya_tahan_baterai
    
    def getTipeKoneksi(self):
        return self.tipe_koneksi
    
    def getRangeKoneksi(self):
        return self.range_koneksi

File: cpp/MouseWireless.cpp
#pragma once
#include "Mouse.cpp"
#include <iostream>
#include <string>
using namespace std;

class MouseWireless : public Mouse
{
private:
    int kapasitas_baterai;
    string daya_tahan_baterai;
    string tipe_koneksi;
    string range_koneksi;

public:
    MouseWireless() {}

    MouseWireless(int kapasitas_baterai, string daya_tahan_baterai, string tipe_koneksi, string range_koneksi, string nama, string bentuk_mouse, string DPI, string warna, int jumlah_tombol, string tipe_koneksi_mouse) : Mouse(nama, bentuk_mouse, DPI, warna, jumlah_tombol, tipe_koneksi_mouse)
    {
        this->kapasitas_baterai = kapasitas_baterai;
        this->daya_tahan_baterai = daya_tahan_baterai;
        this->tipe_koneksi = tipe_koneksi;
        this->range_koneksi = range_koneksi;
    }

    void setKapasitasBaterai(int kapasitas_baterai)
    {
        this->kapasitas_baterai = kapasitas_baterai;
    }

    void setDayaTahanBaterai(string daya_tahan_baterai)
    {
        this->daya_tahan_baterai = daya_tahan_baterai;
    }

    void setTipeKoneksi(string tipe_koneksi)
    {
        this->tipe_koneksi = tipe_koneksi;
    }

    void setRangeKoneksi(string range_koneksi)
    {
        this->range_koneksi = range_koneksi;
    }

    int getKapasitasBaterai()
    {
        return kapasitas_baterai;
    }

    string getDayaTahanBaterai()
    {
        return daya_tahan_baterai;
    }

    string getTipeKoneksi()
    {
        return tipe_koneksi;
    }

    string getRangeKoneksi()
    {
        return range_koneksi;
    }

    ~MouseWireless() {}
};